### PR TITLE
fix: improve errors when running against Konnect with workspace config

### PR DESCRIPTION
**Pull request by GGabriele**

### CODE REVIEW

**Review by pmalek — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +40,17 @@
...
 
+## [v1.13.0]
+
+> Release date: TBD
+
+### Fixes
+
+- Since Konnect doesn't have any concept of workspaces, decK should return
+  a meaningfull error message whenever running against Konnect with a config file
+  having the `_workspace` entry or the `--workspace` flag is used.
```
Comment (hbagdi) [inline CHANGELOG.md:51]:
```suggestion
- Improve error message when workspace is used in Konnect mode
```

A changelog is for the end-user. They need to be notified of changes, not of 'why', 'how', 'when' at this level of detail.

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,6 +40,17 @@
 - [v0.2.0](#v020)
 - [v0.1.0](#v010)
 
+## [v1.13.0]
```
Comment (hbagdi) [inline CHANGELOG.md:43]: What if there is another patch release?
Also, users may not actually read  'Release date: TBD' and get confused.
Please rename this header to 'Unreleased'.

File: cmd/common.go [hunk 3]
```diff
@@ -101,6 +101,13 @@ func syncMain(ctx context.Context, filenames []string, dry bool, parallelism,
 	var kongClient *kong.Client
 	mode := getMode(targetContent)
 	if mode == modeKonnect {
+		if targetContent.Workspace != "" {
+			return fmt.Errorf("_workspace entry found in config file.\n" +
+				"Workspaces are not supported in Konnect")
```
Comment (hbagdi) [inline cmd/common.go:106]: Please make the error more actionable. What is a user supposed to do after this error?
Something along the lines (same applies for the other error):
```suggestion
			return fmt.Errorf("_workspace set in config file.\n" +
				"Workspaces are not supported in Konnect. Please remove '_workspace: foo' from your configuration and try again")
```

**Review by hbagdi — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (GGabriele) [reply CHANGELOG.md]: Roger that, makes sense. Fixed!

File: cmd/common.go [hunk 4]
```diff
@@ -101,6 +101,14 @@ func syncMain(ctx context.Context, filenames []string, dry bool, parallelism,
 	var kongClient *kong.Client
 	mode := getMode(targetContent)
 	if mode == modeKonnect {
+		if targetContent.Workspace != "" {
+			return fmt.Errorf("_workspace set in config file.\n" +
```
Comment (jrsmroz) [inline cmd/common.go:105]: Does it needs to be multi line?

**Comment by hbagdi** [comment]
Please rebase.

**Comment by GGabriele** [comment]
> Please rebase.

done

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,7 +1,10 @@
 # Table of Contents
 
 - [Unreleased](#Unreleased)
+<<<<<<< HEAD
 - [v1.13.0](#v1130)
+=======
+>>>>>>> 9ef8a61 (address comments)
```
Comment (pmalek) [inline CHANGELOG.md:7]: oops

**Review by pmalek — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (GGabriele) [reply CHANGELOG.md:7]: 🤦

File: cmd/common.go [hunk 6]
```diff
@@ -101,6 +101,14 @@ func syncMain(ctx context.Context, filenames []string, dry bool, parallelism,
...
+				"Workspaces are not supported in Konnect. " +
+				"Please remove '_workspace: foo' from your configuration and try again")
```
Comment (hbagdi) [inline cmd/common.go:107]: 'foo' seems to be a copy-paste error. Could you please update the error message to include the workspace that the user has provided?

Comment (GGabriele) [reply cmd/common.go]: done
```
$ ./deck sync --konnect-token $TOKEN_DEV
Error: _workspace set in config file.
Workspaces are not supported in Konnect. Please remove '_workspace: test' from your configuration and try again
```

**Pull request merged**